lib/TokenManager.ts

```diff
@@ -393,8 +495,35 @@ export class TokenManager {
         this.remove(key);
       }
     };
-    this.on('expired', onTokenExpiredHandler);
+    this.on(EVENT_EXPIRED, onTokenExpiredHandler);
 
     setExpireEventTimeoutAll(sdk, tokenMgmtRef, storage);
+
+    // Sync authState cross multiple tabs when localStorage is used as the storageProvider
+    // A StorageEvent is sent to a window when a storage area it has access to is changed 
+    // within the context of another document.
+    // https://developer.mozilla.org/en-US/docs/Web/API/StorageEvent
+    window.addEventListener('storage', ({ key, newValue, oldValue }: StorageEvent) => {
+      const handleCrossTabsStorageChange = () => {
+        this._resetExpireEventTimeoutAll();
+        this._emitEventsForCrossTabsStorageUpdate(newValue, oldValue);
+      };
+
+      // Skip if:
+      // not from localStorage.clear (event.key is null)
+      // event.key is not the storageKey
+      // oldValue === newValue
+      if (key && (key !== options.storageKey || newValue === oldValue)) {
+        return;
+      }
+
+      // LocalStorage cross tabs update is not synced in IE, set a 1s timer to read latest value
+      // https://stackoverflow.com/questions/24077117/localstorage-in-win8-1-ie11-does-not-synchronize
+      if (isIE11OrLess()) {
```

Comment:
What do you think about having this function always called in a setTimeout, but have the time value be configurable. We can set it by default to 0 unless IE11 where we set it by default to 1000. But we could also manually set it to a value in our test/sample apps to verify that the logic works as we expect without needing to fake IE11